### ISSUE TITLE
API ObjectDetection size of input images

### ISSUE THREAD

**Issue by edrti** [tensorflow/models#8622]
Hi,
I have a data set in which every images is in different size,
is it possible to train the dataset via the tensorflow objectdetection api?

tnx

**Comment by jaykshirsagar05** [comment]
You can preprocess and transform your dataset in a uniform manner before feeding it into the model.
This might help you
[https://www.tensorflow.org/guide/data](url) or [https://www.tensorflow.org/tfx/tutorials/transform/simple](url)

**Comment by vijaygill** [comment]
> You can preprocess and transform your dataset in a uniform manner before feeding it into the model.
> This might help you
> [https://www.tensorflow.org/guide/data](url) or [https://www.tensorflow.org/tfx/tutorials/transform/simple](url)

The links you posted need to be fixed. The text and the actual destination are not the same.

**Comment by jaykshirsagar05** [comment]
Links are updated you can check it once again.

**Comment by edrti** [comment]
which preprocess should i do? resize all of them to the same size?

**Comment by jaykshirsagar05** [comment]
Yes, you should resize all the images according to the model's size requirements.  Also, you can use tf.random_crop or tf.central_crop according to your need.

**Comment by edrti** [comment]
thanks.
so i am confused, resize will hurt the image quality, i thinking padding each image with black pixels in order to achieve specific size, like this:[
![cropped_99_img_0186](https://user-images.githubusercontent.com/26519669/83759434-35fab600-a67c-11ea-81c5-481cd0be4daa.jpg)
![cropped_97_img_0186](https://user-images.githubusercontent.com/26519669/83759437-372be300-a67c-11ea-9fc1-85d5691d7f61.jpg)

]
what is better? resize or padding?

**Comment by jaykshirsagar05** [comment]
In my opinion, padding will create a redundant black part which is of no use and model will have to learn that unnecessarily so you might use resize and crop only for more info you can visit over here https://stackoverflow.com/questions/47697622/cnn-image-resizing-vs-padding-keeping-aspect-ratio-or-not
